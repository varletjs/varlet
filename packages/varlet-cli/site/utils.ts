import { get } from 'lodash-es'
import { StyleProvider, type Theme, withSiteConfigNamespace } from '@varlet/cli/client'

export interface Menu {
  doc: string
  text: Record<string, string>
  type: MenuTypes
  useMobile?: boolean
}

export enum MenuTypes {
  TITLE = 1,
  COMPONENT = 2,
  DOCUMENTATION = 3,
}

export function removeEmpty(object: Record<string, string> = {}) {
  return Object.keys(object).reduce((record: Record<string, string>, key) => {
    const value = object[key]
    value && (record[key] = value)
    return record
  }, {})
}

export function isPhone() {
  return /Android|webOS|iPhone|iPod|BlackBerry|Pad/i.test(navigator.userAgent)
}

export function inIframe() {
  return window.self !== window.top
}

export function setTheme(config: Record<string, any>, name: Theme) {
  const styleVars = withSiteConfigNamespace(get(config, name, {}))

  StyleProvider(styleVars)
}

export function utoa(data: string): string {
  return btoa(unescape(encodeURIComponent(data)))
}
