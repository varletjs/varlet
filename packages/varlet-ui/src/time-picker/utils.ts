import { hours24, hoursAmpm } from './props'
import type { AmPm, AllowedTime } from './props'
import { toNumber } from '@varlet/shared'

type DisableProps = {
  time: number
  format: string
  ampm: AmPm | undefined
  hour: string
  minute: number
  max: string | undefined
  min: string | undefined
  disableHour: Array<string>
  allowedTime: AllowedTime | undefined
}

export const notConvert = (format: string, ampm: AmPm | undefined): boolean => format === '24hr' || ampm === 'am'

export const convertHour = (format: string, ampm: AmPm | undefined, hour: string) => {
  const index = hoursAmpm.findIndex((hourAmpm) => toNumber(hourAmpm) === toNumber(hour))
  const getHour = notConvert(format, ampm) ? hour : hours24[index]

  return {
    hourStr: getHour,
    hourNum: toNumber(getHour),
  }
}

export const getNumberTime = (time: string) => {
  const [hour, minute, second] = time.split(':')

  return {
    hour: toNumber(hour),
    minute: toNumber(minute),
    second: toNumber(second),
  }
}

export const getIsDisableMinute = (values: Omit<DisableProps, 'minute'>): boolean => {
  const { time, format, ampm, hour, max, min, disableHour } = values
  const { hourStr, hourNum } = convertHour(format, ampm, hour)
  let isBetweenMinMax = false
  let isAllow = false

  if (disableHour.includes(hourStr)) return true

  if (max && !min) {
    const { hour: maxHour, minute: maxMinute } = getNumberTime(max)
    isBetweenMinMax = maxHour === hourNum && time > maxMinute
  }

  if (!max && min) {
    const { hour: minHour, minute: minMinute } = getNumberTime(min)

    isBetweenMinMax = minHour === hourNum && time < minMinute
  }

  if (max && min) {
    const { hour: maxHour, minute: maxMinute } = getNumberTime(max)
    const { hour: minHour, minute: minMinute } = getNumberTime(min)

    isBetweenMinMax = (minHour === hourNum && time < minMinute) || (maxHour === hourNum && time > maxMinute)
  }

  if (values.allowedTime?.minutes) isAllow = values.allowedTime?.minutes(time)

  return isBetweenMinMax || isAllow
}

export const getIsDisableSecond = (values: DisableProps): boolean => {
  const { time, format, ampm, hour, minute, max, min, disableHour } = values
  const { hourStr, hourNum } = convertHour(format, ampm, hour)
  let isBetweenMinMax = false
  let isAllow = false

  if (disableHour.includes(hourStr)) return true

  if (max && !min) {
    const { hour: maxHour, minute: maxMinute, second: maxSecond } = getNumberTime(max)

    isBetweenMinMax = (maxHour === hourNum && maxMinute < minute) || (maxMinute === minute && time > maxSecond)
  }

  if (!max && min) {
    const { hour: minHour, minute: minMinute, second: minSecond } = getNumberTime(min)

    isBetweenMinMax = (minHour === hourNum && minMinute > minute) || (minMinute === minute && time > minSecond)
  }

  if (max && min) {
    const { hour: maxHour, minute: maxMinute, second: maxSecond } = getNumberTime(max)
    const { hour: minHour, minute: minMinute, second: minSecond } = getNumberTime(min)

    isBetweenMinMax =
      (maxHour === hourNum && maxMinute < minute) ||
      (minHour === hourNum && minMinute > minute) ||
      (maxHour === hourNum && maxMinute === minute && time > maxSecond) ||
      (minHour === hourNum && minMinute === minute && time < minSecond)
  }

  if (values.allowedTime?.seconds) isAllow = values.allowedTime?.seconds(time)

  return isBetweenMinMax || isAllow
}
