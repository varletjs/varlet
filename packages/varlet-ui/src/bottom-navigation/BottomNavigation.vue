<template>
  <div
    :class="classes(n(), [fixed, n('--fixed')], [border, n('--border')])"
    ref="bottomNavigationDom"
    :style="`z-index:${zIndex}`"
  >
    <slot></slot>

    <var-button
      v-if="$slots.fab"
      :class="classes(n('fab'), [length % 2, n('--fab-right'), n('--fab-center')])"
      var-bottom-navigation__fab
      @click="handleFabClick"
      v-bind="fabProps"
      round
    >
      <slot name="fab"></slot>
    </var-button>
  </div>
</template>

<script lang="ts">
import VarButton from '../button'
import { defineComponent, ref, computed, onMounted, onUpdated, watch } from 'vue'
import { props } from './props'
import { useBottomNavigationItems } from './provide'
import { createNamespace, call } from '../utils/components'
import { isNumber } from '../utils/shared'
import type { BottomNavigationProvider } from './provide'
import type { BottomNavigationItemProvider } from '../bottom-navigation-item/provide'
import type { Ref, ComputedRef } from 'vue'

const { n, classes } = createNamespace('bottom-navigation')
const { n: nItem } = createNamespace('bottom-navigation-item')

const RIGHT_HALF_SPACE_CLASS = nItem('--right-half-space')
const LEFT_HALF_SPACE_CLASS = nItem('--left-half-space')
const RIGHT_SPACE_CLASS = nItem('--right-space')
const defaultFabProps = {
  type: 'primary',
}

export default defineComponent({
  name: 'VarBottomNavigation',
  components: { VarButton },
  props,
  setup(props, { slots }) {
    const bottomNavigationDom: Ref<HTMLElement | null> = ref(null)
    const active: ComputedRef<number | string | undefined> = computed(() => props.active)
    const activeColor: ComputedRef<string | undefined> = computed(() => props.activeColor)
    const inactiveColor: ComputedRef<string | undefined> = computed(() => props.inactiveColor)
    const fabProps = ref({})
    const { length, bottomNavigationItems, bindBottomNavigationItem } = useBottomNavigationItems()

    const matchBoundary = (): void => {
      if (length.value === 0 || matchName() || matchIndex()) {
        return
      }

      handleActiveIndex()
    }

    const matchName = (): BottomNavigationItemProvider | undefined => {
      return bottomNavigationItems.find(({ name }: BottomNavigationItemProvider) => {
        return active.value === name.value
      })
    }

    const matchIndex = (): BottomNavigationItemProvider | undefined => {
      return bottomNavigationItems.find(({ index }: BottomNavigationItemProvider) => {
        return active.value === index.value
      })
    }

    const handleActiveIndex = () => {
      if (!isNumber(active.value)) {
        return
      }

      if (active.value < 0) {
        call(props['onUpdate:active'], 0)
      } else if (active.value > length.value - 1) {
        call(props['onUpdate:active'], length.value - 1)
      }
    }

    const onToggle = (changedValue: number | string) => {
      props.onBeforeChange ? handleBeforeChange(changedValue) : handleChange(changedValue)
    }

    const handleBeforeChange = (changedValue: number | string) => {
      Promise.resolve(call(props.onBeforeChange, changedValue)).then((res) => res && handleChange(changedValue))
    }

    const handleChange = (changedValue: number | string) => {
      call(props['onUpdate:active'], changedValue)
      call(props.onChange, changedValue)
    }

    const removeMarginClass = () => {
      const bottomNavigationItems: Element[] = getBottomNavigationItems()
      bottomNavigationItems.forEach((dom: Element) => {
        dom.classList.remove(RIGHT_HALF_SPACE_CLASS, LEFT_HALF_SPACE_CLASS, RIGHT_SPACE_CLASS)
      })
    }

    const addMarginClass = (length: number) => {
      const bottomNavigationItems: Element[] = getBottomNavigationItems()
      const itemsNum = bottomNavigationItems.length
      const isEven = length % 2 === 0

      bottomNavigationItems.forEach((bottomNavigationItem: Element, i: number) => {
        handleMarginClass(isEven, bottomNavigationItem, i, itemsNum)
      })
    }

    const handleMarginClass = (isEven: boolean, dom: Element, i: number, length: number) => {
      const isLast = i === length - 1
      if (!isEven && isLast) {
        dom.classList.add(RIGHT_SPACE_CLASS)
        return
      }

      const isFabLeft = i === length / 2 - 1
      const isFabRight = i === length / 2
      if (isFabLeft) {
        dom.classList.add(RIGHT_HALF_SPACE_CLASS)
      } else if (isFabRight) {
        dom.classList.add(LEFT_HALF_SPACE_CLASS)
      }
    }

    const getBottomNavigationItems = () => {
      return Array.from(bottomNavigationDom.value!.querySelectorAll(`.${nItem()}`))
    }

    const handleFabClick = () => {
      call(props.onFabClick)
    }

    const bottomNavigationProvider: BottomNavigationProvider = {
      active,
      activeColor,
      inactiveColor,
      onToggle,
    }

    bindBottomNavigationItem(bottomNavigationProvider)

    watch(() => length.value, matchBoundary)

    watch(
      () => props.fabProps,
      (newValue) => {
        fabProps.value = { ...defaultFabProps, ...newValue }
      },
      { immediate: true, deep: true }
    )

    onMounted(() => {
      if (!slots.fab) {
        return
      }

      addMarginClass(length.value)
    })

    onUpdated(() => {
      removeMarginClass()

      if (!slots.fab) {
        return
      }

      addMarginClass(length.value)
    })

    return {
      n,
      classes,
      length,
      bottomNavigationDom,
      handleFabClick,
      fabProps,
    }
  },
})
</script>

<style lang="less">
@import '../styles/common';
@import '../button/button';
@import './bottomNavigation';
</style>
