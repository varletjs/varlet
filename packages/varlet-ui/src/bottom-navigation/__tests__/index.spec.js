import example from '../example'
import BottomNavigation from '..'
import VarBottomNavigation from '../BottomNavigation'
import VarBottomNavigationItem from '../../bottom-navigation-item/BottomNavigationItem'
import { createApp } from 'vue'
import { mount } from '@vue/test-utils'
import { delay, trigger } from '../../utils/jest'

test('test bottom-navigation example', () => {
  const wrapper = mount(example)
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})

test('test bottom-navigation plugin', () => {
  const app = createApp({}).use(BottomNavigation)
  expect(app.component(BottomNavigation.name)).toBeTruthy()
})

test('test bottom-navigation before-change prevent switch', async () => {
  const handleBeforeChange = jest.fn(() => false)
  const Wrapper = {
    components: {
      [VarBottomNavigation.name]: VarBottomNavigation,
      [VarBottomNavigationItem.name]: VarBottomNavigationItem,
    },
    data: () => ({
      active: 0,
    }),
    methods: {
      handleBeforeChange,
    },
    template: `
    <var-bottom-navigation v-model:active ="active" @before-change="handleBeforeChange">
      <var-bottom-navigation-item label="标签" icon="home" />
      <var-bottom-navigation-item label="标签" icon="magnify" />
      <var-bottom-navigation-item label="标签" icon="heart" />
      <var-bottom-navigation-item label="标签" icon="account-circle" />
    </var-bottom-navigation>
    `,
  }
  const wrapper = mount(Wrapper)
  await delay(16)

  const bottomNavigationItem = wrapper.findAll('.var-bottom-navigation-item')[1]
  await trigger(bottomNavigationItem, 'click')

  expect(handleBeforeChange).toHaveBeenCalledTimes(1)
  expect(wrapper.vm.active).toBe(0)

  wrapper.unmount()
})

test('test bottom-navigation before-change return promise', async () => {
  const handleBeforeChange = jest.fn(
    () =>
      new Promise((resolve) => {
        setTimeout(() => {
          resolve(true)
        }, 500)
      })
  )
  const Wrapper = {
    components: {
      [VarBottomNavigation.name]: VarBottomNavigation,
      [VarBottomNavigationItem.name]: VarBottomNavigationItem,
    },
    data: () => ({
      active: 0,
    }),
    methods: {
      handleBeforeChange,
    },
    template: `
    <var-bottom-navigation v-model:active ="active" @before-change="handleBeforeChange">
      <var-bottom-navigation-item label="标签" icon="home" />
      <var-bottom-navigation-item label="标签" icon="magnify" />
      <var-bottom-navigation-item label="标签" icon="heart" />
      <var-bottom-navigation-item label="标签" icon="account-circle" />
    </var-bottom-navigation>
    `,
  }
  const wrapper = mount(Wrapper)
  await delay(16)

  const bottomNavigationItem = wrapper.findAll('.var-bottom-navigation-item')[1]
  await trigger(bottomNavigationItem, 'click')
  expect(wrapper.vm.active).toBe(0)

  await delay(600)
  expect(handleBeforeChange).toHaveBeenCalledTimes(1)
  expect(wrapper.vm.active).toBe(1)

  wrapper.unmount()
})

test('test bottom-navigation change event', async () => {
  let dummy
  const handleChange = jest.fn((active) => {
    dummy = active
  })
  const Wrapper = {
    components: {
      [VarBottomNavigation.name]: VarBottomNavigation,
      [VarBottomNavigationItem.name]: VarBottomNavigationItem,
    },
    data: () => ({
      active: 0,
    }),
    methods: {
      handleChange,
    },
    template: `
    <var-bottom-navigation v-model:active ="active" @change="handleChange">
      <var-bottom-navigation-item label="标签" icon="home" />
      <var-bottom-navigation-item label="标签" icon="magnify" />
      <var-bottom-navigation-item label="标签" icon="heart" />
      <var-bottom-navigation-item label="标签" icon="account-circle" />
    </var-bottom-navigation>
    `,
  }
  const wrapper = mount(Wrapper)
  await delay(16)

  const bottomNavigationItem = wrapper.findAll('.var-bottom-navigation-item')[1]
  await trigger(bottomNavigationItem, 'click')

  expect(handleChange).toHaveBeenCalledTimes(1)
  expect(dummy).toBe(1)

  wrapper.unmount()
})

test('test bottom-navigation-item click event', async () => {
  let dummy
  const handleClick = jest.fn((active) => {
    dummy = active
  })
  const Wrapper = {
    components: {
      [VarBottomNavigation.name]: VarBottomNavigation,
      [VarBottomNavigationItem.name]: VarBottomNavigationItem,
    },
    data: () => ({
      active: 0,
    }),
    methods: {
      handleClick,
    },
    template: `
    <var-bottom-navigation v-model:active ="active">
      <var-bottom-navigation-item label="标签" icon="home" />
      <var-bottom-navigation-item @click="handleClick" label="标签" icon="magnify" />
      <var-bottom-navigation-item label="标签" icon="heart" />
      <var-bottom-navigation-item label="标签" icon="account-circle" />
    </var-bottom-navigation>
    `,
  }
  const wrapper = mount(Wrapper)
  await delay(16)

  const bottomNavigationItem = wrapper.findAll('.var-bottom-navigation-item')[1]
  await trigger(bottomNavigationItem, 'click')

  expect(handleClick).toHaveBeenCalledTimes(1)
  expect(dummy).toBe(1)

  wrapper.unmount()
})

test('test BottomNavigation relation BottomNavigationItems', async () => {
  const Wrapper = {
    components: {
      [VarBottomNavigation.name]: VarBottomNavigation,
      [VarBottomNavigationItem.name]: VarBottomNavigationItem,
    },
    data: () => ({
      active: 0,
    }),
    template: `
    <var-bottom-navigation v-model:active ="active">
      <var-bottom-navigation-item label="标签" icon="home" />
      <var-bottom-navigation-item label="标签" icon="magnify" />
      <var-bottom-navigation-item label="标签" icon="heart" />
      <var-bottom-navigation-item label="标签" icon="account-circle" />
    </var-bottom-navigation>
    `,
  }
  const wrapper = mount(Wrapper)
  await delay(100)

  expect(wrapper.html()).toMatchSnapshot()
  await wrapper.setData({ active: 1 })

  await delay(50)
  expect(wrapper.html()).toMatchSnapshot()

  wrapper.unmount()
})

test('fab button', async () => {
  const Wrapper = {
    components: {
      [VarBottomNavigation.name]: VarBottomNavigation,
      [VarBottomNavigationItem.name]: VarBottomNavigationItem,
    },
    data: () => ({
      active: 0,
    }),
    template: `
    <var-bottom-navigation v-model:active ="active">
      <template #fab>
        <div class="tab_example" />
      </template>
      <var-bottom-navigation-item label="标签" icon="home" />
      <var-bottom-navigation-item label="标签" icon="magnify" />
      <var-bottom-navigation-item label="标签" icon="heart" />
    </var-bottom-navigation>
    `,
  }
  const wrapper = mount(Wrapper)
  await delay(100)

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.unmount()
})
