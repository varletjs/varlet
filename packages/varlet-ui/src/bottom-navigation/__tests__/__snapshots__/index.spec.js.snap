// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fab button 1`] = `
"<div class=\\"var-bottom-navigation\\" style=\\"z-index: 1;\\"><button class=\\"var-bottom-navigation-item var-bottom-navigation-item--active\\"><i class=\\"var-icon var-icon--set var-icon-home var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-magnify var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item var-bottom-navigation-item--right-space\\"><i class=\\"var-icon var-icon--set var-icon-heart var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-button var--box var-button--normal var--inline-flex var-button--primary var-elevation--2 var-button--round var-bottom-navigation__fab var-bottom-navigation--fab-right\\" var-bottom-navigation__fab=\\"\\">
    <!--v-if-->
    <div class=\\"var-button__content\\">
      <div class=\\"tab_example\\"></div>
    </div>
  </button></div>"
`;

exports[`test BottomNavigation relation BottomNavigationItems 1`] = `
"<div class=\\"var-bottom-navigation\\" style=\\"z-index: 1;\\"><button class=\\"var-bottom-navigation-item var-bottom-navigation-item--active\\"><i class=\\"var-icon var-icon--set var-icon-home var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-magnify var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-heart var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-account-circle var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button>
  <!--v-if-->
</div>"
`;

exports[`test BottomNavigation relation BottomNavigationItems 2`] = `
"<div class=\\"var-bottom-navigation\\" style=\\"z-index: 1;\\"><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-home var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item var-bottom-navigation-item--active\\"><i class=\\"var-icon var-icon--set var-icon-magnify var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-heart var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-account-circle var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button>
  <!--v-if-->
</div>"
`;

exports[`test bottom-navigation example 1`] = `
"<div class=\\"app-type\\">基本使用</div>
<div class=\\"var-bottom-navigation\\" style=\\"z-index: 1;\\"><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-home var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-magnify var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-heart var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-account-circle var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button>
  <!--v-if-->
</div>
<div class=\\"app-type\\">通过名称匹配</div>
<div class=\\"var-bottom-navigation\\" style=\\"z-index: 1;\\"><button class=\\"var-bottom-navigation-item var-bottom-navigation-item--active\\"><i class=\\"var-icon var-icon--set var-icon-home var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-magnify var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-heart var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-account-circle var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button>
  <!--v-if-->
</div>
<div class=\\"app-type\\">徽标提示</div>
<div class=\\"var-bottom-navigation\\" style=\\"z-index: 1;\\"><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-home var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-magnify var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <div class=\\"var-badge var--box\\">
      <transition-stub><span class=\\"var-bottom-navigation-item__badge var-badge__content var-badge--danger var-badge__dot var-badge__dot--right\\" var-bottom-navigation-item-cover=\\"\\"><span></span></span></transition-stub>
    </div><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-heart var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <div class=\\"var-badge var--box\\">
      <transition-stub><span class=\\"var-bottom-navigation-item__badge var-badge__content var-badge--primary\\" var-bottom-navigation-item-cover=\\"\\"><span>66</span></span></transition-stub>
    </div><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-account-circle var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button>
  <!--v-if-->
</div>
<div class=\\"app-type\\">自定义颜色</div>
<div class=\\"var-bottom-navigation\\" style=\\"z-index: 1;\\"><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-home var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-magnify var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-heart var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-account-circle var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button>
  <!--v-if-->
</div>
<div class=\\"app-type\\">监听切换事件</div>
<div class=\\"var-bottom-navigation\\" style=\\"z-index: 1;\\"><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-home var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-magnify var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-heart var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-account-circle var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button>
  <!--v-if-->
</div>
<div class=\\"app-type\\">监听点击事件</div>
<div class=\\"var-bottom-navigation\\" style=\\"z-index: 1;\\"><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-home var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-magnify var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-heart var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-account-circle var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button>
  <!--v-if-->
</div>
<div class=\\"app-type\\">悬浮按钮</div>
<div class=\\"var-bottom-navigation\\" style=\\"z-index: 1; margin-top: 10px;\\"><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-home var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item var-bottom-navigation-item--right-half-space\\"><i class=\\"var-icon var-icon--set var-icon-magnify var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item var-bottom-navigation-item--left-half-space\\"><i class=\\"var-icon var-icon--set var-icon-heart var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button><button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-icon--set var-icon-bell var-bottom-navigation-item__icon\\" style=\\"transition: transform 0ms;\\" var-bottom-navigation-item-cover=\\"\\"></i>
    <!--v-if--><span class=\\"var-bottom-navigation-item__label\\">标签</span>
  </button>
  <!--v-if--><button class=\\"var-button var--box var-button--normal var--inline-flex var-button--primary var-elevation--2 var-button--round var-bottom-navigation__fab var-bottom-navigation--fab-center\\" var-bottom-navigation__fab=\\"\\">
    <!--v-if-->
    <div class=\\"var-button__content\\"><i class=\\"var-icon var-icon--set var-icon-heart\\" style=\\"transition: transform 0ms;\\"></i></div>
  </button>
</div>"
`;
